### PR TITLE
Add Review requests and referrals (x) link behind feature flag

### LINKED ISSUE #95300: Remove Pending Tab

### PR DESCRIPTION
**Pull request by juancstlm-a6**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
- Hides the Pending tab in appointments list and adds a new link to view requests and referrals hidden under a feature flag `vaOnlineSchedulingCCDirectScheduling` 

## Related issue(s)
department-of-veterans-affairs/va.gov-team#95300

## Testing done
- Previously Appointments list had a "Pending" tab that would list the number of pending appointments
- To test this there must be a pending appointment and have `vaOnlineSchedulingCCDirectScheduling` feature flag on
- Added unit tests to ensure only the above scenario triggers the new UI

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| 📱  |Screen Shot 2024-11-06 at 19 47 17|Screen Shot 2024-11-08 at 09 54 24|
| 🖥️ |Screenshot 2024-11-06 at 7 47 08 PM|Screenshot 2024-11-08 at 9 54 38 AM|

## What areas of the site does it impact?

- Appointments list feature in VAOS

## Acceptance criteria
- [ ] remove pending tab link when feature is on
- [ ] Add `Review requests and referrals` link if feature is on
- [ ] add unit test for added functionality

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback
- Link to view requests and referrals will be updated in a follow up PR that adds that functionality

### CODE REVIEW

**Review by rmessina1010 — DISMISSED** [review]
LGTM

File: src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import PropTypes from 'prop-types';
 import {
   selectFeatureBreadcrumbUrlUpdate,
   selectFeatureCCDirectScheduling,
+  selectFeatureCCReviewRequestAndReferrals,
```
Comment (brianseek) [inline src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx:12]: use the existing feature of `vaOnlineSchedulingCCDirectScheduling`

File: src/applications/vaos/appointment-list/components/ReviewRequestsAndReferrals.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import classNames from 'classnames';
+
+export default function ReviewRequestsAndReferrals() {
```
Comment (brianseek) [inline src/applications/vaos/appointment-list/components/ReviewRequestsAndReferrals.jsx:4]: I would name this a little more specifically. Something like `RequestsAndReferralsLink`. I feel like that would make it quick to identify in the list.

**Review by brianseek — CHANGES_REQUESTED** [review]
Nice work on this stuff @juancstlm-a6! A couple little things to update. Also with the link being its own component, it should have a unit test for code coverage sake. Even though there isn't really anything happening in there. 

Another option around that would be to not make it a component and add the markup to the page behind the feature. I'll leave that up to you though.

File: src/applications/vaos/appointment-list/components/ReviewRequestsAndReferrals.jsx (hunk 2 above)
Comment (juancstlm-a6) [reply src/applications/vaos/appointment-list/components/ReviewRequestsAndReferrals.jsx]: Moving this from a component to the mark up being in the page so no longer relevant, but I agree with comments.

File: src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx [hunk 3]
```diff
@@ -170,9 +164,28 @@ export default function AppointmentsPage() {
           <ReferralAppLink linkText="Review and manage your appointment notifications" />
         </div>
       )}
-      {featureCCReviewRequestAndReferrals && <ReviewRequestsAndReferrals />}
+      {featureCCDirectScheduling && (
+        <div
+          className={classNames(
+            'vads-u-padding-y--3',
+            'vads-u-margin-bottom--3',
+            'vads-u-margin-top--1',
```
Comment (brianseek) [inline src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx:172]: One last little thing, looks like this is a little tight. I think vad-u-margin-top--3 looks close to the mock.

**Review by brianseek — CHANGES_REQUESTED** [review]
One more thing. Little margin tweak

File: src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx (hunk 3 above)
Comment (brianseek) [reply src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx:172]: scratch that, it looks off when there are two actions but not one

Comment (brianseek) [reply src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx:172]: Screenshot 2024-11-08 at 9 52 44 AM

Comment (brianseek) [reply src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx:172]: should probably be fixed elsewhere, so you can ignore this for this ticket

**Review by brianseek — APPROVED** [review]

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**